[Parse] Properly reject labeled yield

hamishknight opened the pull request:
Instead of asserting, emit a diagnostic with a fix-it to remove the label.

SR-15066
rdar://82132821

hamishknight commented:
@swift-ci please test

LucianoPAlmeida commented on include/swift/AST/DiagnosticsParse.def:
```diff
@@ -1030,6 +1030,8 @@ ERROR(expected_await_not_async,none,
 ERROR(expected_expr_yield,PointsToFirstBadToken,
       "expected expression in 'yield' statement", ())
+ERROR(unexpected_label_yield,none,
+      "unexpected label in yield", ())
```
Hey @hamishknight, I wonder if it would make sense to have it like `'yield'` or even like the above `in 'yield' statement` maybe?

hamishknight replied:
Good catch! Yeah definitely best to be consistent here

hamishknight committed:
[Parse] Remove dead condition

We used to avoid parsing labels for yield stmts
by passing in `SyntaxKind::YieldStmt` and relying
on this condition. However we no longer do so, we
pass `SyntaxKind::ExprList` instead.

hamishknight committed:
[Parse] Properly reject labeled yield

Instead of asserting, emit a diagnostic with a
fix-it to remove the label.

SR-15066
rdar://82132821

hamishknight commented:
@swift-ci please smoke test
